tests/filesystem/FileTest.php

```diff
@@ -187,6 +187,30 @@ public function testSetNameWithInvalidExtensionDoesntChangeFilesystem() {
 		}
 	}
 
+	/**
+	 * Uses fixtures Folder.folder1 and File.setfromname
+	 * @dataProvider setNameFileProvider
+	 */
+	public function testSetNameAddsUniqueSuffixWhenFilenameAlreadyExists($name, $expected)
+	{
+		$duplicate = new Folder;
+		$duplicate->setName($name);
+		$duplicate->write();
+
+		$this->assertSame($expected, $duplicate->Name);
+	}
+
+	/**
+	 * @return array[]
+	 */
+	public function setNameFileProvider()
```

Comment:
non-test methods should be `protected` or `private`

Reply:
It's a provider - they need to be public don't they?

Reply:
I don't believe so.

Reply:
[They do](https://phpunit.de/manual/5.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers) in PHPUnit 5.x at least, may have changed in 6.x